guides/v2.2/extension-dev-guide/framework/serializer.md

```diff
@@ -113,3 +113,36 @@ public function loadDataFromCache()
 ...
 
 {% endhighlight %}
+
+## Backward Compatibility Note
+
+The `SerializerInterface` interface and its implementations only exist since Magento version 2.2.  
+Because of this, it is not posssible to use these classes in code that has to be compatible with Magento 2.1 or 2.0.  
+
+In code that is compatible with earlier versions of Magento 2, constructor dependency injection can not be used to get an instance of `SerializerInterface`.  
+Instead, a runtime check if the `SerializerInterface` definition exists can made, and if it does, it can be instantiated by directly accessing the object manager using a static method. Alternatively a check against the Magento 2 version or the `magento/framework` composer package version would work, too.  
+If the interface does not exist or an earlier version of Magento 2 is being executed, the appropriate native PHP serialization function has to be called, e.g. `\serialize()` or `\json_encode()`, depending on the usercase.  
+
+Here is an example:
+
+{% highlight php startinline %}
+
+```
+/**
+ * @param mixed $data
+ * @return string
+ */
+ private function serialize($data)
+ {
+    if (class_exists(\Magento\Framework\Serialize\SerializerInterface::class)) {
```

Comment:
Wouldn't this throw a php warning when the class `\Magento\Framework\Serialize\SerializerInterface` doesn't exist in 2.1 or 2.0? It would probably be better to use a string representation of the class name I think?

Reply:
No, it does not throw an exception or raise a warning.
The PHP "compiler" will apply namespace rules to `::class` constants and then uses it as a string, so there is literally zero difference between
`if (class_exists(\Magento\Framework\Serialize\SerializerInterface::class))`
and
`if (class_exists('\Magento\Framework\Serialize\SerializerInterface'))`
except that the former allows static code analysis to do a better job.

Reply:
Hi @Vinai, I just tested this, you are absolutely correct, learned something new today, thanks! :)